appium-inspector: 2024.12.1 -> 2025.3.1

NyCodeGHG opened the pull request:
Diff: https://github.com/appium/appium-inspector/compare/v2024.12.1...v2025.3.1
Changelog: https://github.com/appium/appium-inspector/releases/tag/v2025.3.1


<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

NyCodeGHG committed:
appium-inspector: 2024.12.1 -> 2025.3.1

Diff: https://github.com/appium/appium-inspector/compare/v2024.12.1...v2025.3.1
Changelog: https://github.com/appium/appium-inspector/releases/tag/v2025.3.1

donovanglover commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha)

Command: `nixpkgs-review pr 391819`

Logs: https://github.com/donovanglover/nixpkgs-review-gha/actions/runs/14021491265


---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>appium-inspector</li>
  </ul>
</details>


---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>appium-inspector</li>
  </ul>
</details>


---
### `x86_64-darwin`
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>appium-inspector</li>
  </ul>
</details>


---
### `aarch64-darwin`
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>appium-inspector</li>
  </ul>
</details>

donovanglover commented:
Relevant part of logs:

```rust
appium-inspector>   ⨯ open /private/tmp/nix-build-appium-inspector-2025.3.1.drv-0/source/release/mac/Electron.app/Contents/Frameworks/Electron Helper (GPU).app/Contents/Info.plist: permission denied
appium-inspector>   ⨯ /private/tmp/nix-build-appium-inspector-2025.3.1.drv-0/source/node_modules/app-builder-bin/mac/app-builder_amd64 process failed ERR_ELECTRON_BUILDER_CANNOT_EXECUTE
appium-inspector> Exit code:
appium-inspector> 1  failedTask=build stackTrace=Error: /private/tmp/nix-build-appium-inspector-2025.3.1.drv-0/source/node_modules/app-builder-bin/mac/app-builder_amd64 process failed ERR_ELECTRON_BUILDER_CANNOT_EXECUTE
appium-inspector> Exit code:
appium-inspector> 1
appium-inspector>     at ChildProcess.<anonymous> (/private/tmp/nix-build-appium-inspector-2025.3.1.drv-0/source/node_modules/builder-util/src/util.ts:255:14)
appium-inspector>     at Object.onceWrapper (node:events:633:26)
appium-inspector>     at ChildProcess.emit (node:events:518:28)
appium-inspector>     at maybeClose (node:internal/child_process:1101:16)
appium-inspector>     at Socket.<anonymous> (node:internal/child_process:456:11)
appium-inspector>     at Socket.emit (node:events:518:28)
appium-inspector>     at Pipe.<anonymous> (node:net:351:12)
```

donovanglover commented:
Try changing `inherit (electron.meta) platforms;` to `platforms = lib.platforms.linux;` (ofborg tries to build on darwin too)

NyCodeGHG commented:
Oh I see what the problem is, `electron_34.meta.platforms` evaluates to different platforms, depending on what hostPlatform you're evaluating for.
```
$ nix-instantiate --eval -A electron_34.meta.platforms --system aarch64-darwin
[ "x86_64-darwin" "x86_64-linux" "armv7l-linux" "aarch64-linux" "aarch64-darwin" ]
$ nix-instantiate --eval -A electron_34.meta.platforms --system x86_64-linux
[ "aarch64-linux" "armv5tel-linux" "armv6l-linux" "armv7a-linux" "armv7l-linux" "i686-linux" "loongarch64-linux" "m68k-linux" "microblaze-linux" "microblazeel-linux" "mips-linux" "mips64-linux" "mips64el-linux" "mipsel-linux" "powerpc64-linux" "powerpc64le-linux" "riscv32-linux" "riscv64-linux" "s390-linux" "s390x-linux" "x86_64-linux" ]
```

NyCodeGHG committed:
appium-inspector: restrict platforms to linux

electron.meta.platforms evals to different values depending on which
system you eval for. This confuses hydra so we stop doing that since
this package only builds for linux anyways.
